gcc/rust/resolve/rust-ast-resolve-pattern.h

```diff
@@ -25,6 +25,14 @@
 namespace Rust {
 namespace Resolver {
 
+enum PatternBoundCtx
+{
+  Product,
```

Comment:
can you comment what each of these mean? Its not clear to me what they represent